docs/azure/sdk/authentication/create-token-credentials-from-configuration.md

```diff
@@ -0,0 +1,135 @@
+
+Azure service clients registered with the `AddAzureClients` method are automatically configured with an instance of `DefaultAzureCredential`. You can override the global `DefaultAzureCredential` using credential values from configuration files when registering a client:
+
+```csharp
+builder.Services.AddAzureClients(clientBuilder =>
+{
+    // Register BlobServiceClient using credentials from appsettings.json
+    clientBuilder.AddBlobServiceClient(builder.Configuration.GetSection("Storage"));
+
+    // Register ServiceBusClient using the fallback DefaultAzureCredential credentials
+    clientBuilder.AddServiceBusClientWithNamespace(
+        "<your_namespace>.servicebus.windows.net");
+});
+```
+
+The associated _appsettings.json_ file:
+
+```json
+"Storage": {
+    "serviceUri": "<service_uri>",
+    "credential": "managedidentity",
+    "clientid":  "<clientId>"
+}
+```
+
+### Create a `ManagedIdentityCredential` type
+
+You can create both user-assigned and system-assigned managed identities using configuration values. Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ManagedIdentityCredential?displayProperty=fullName>.
+
+#### User-assigned identities
+
+1. Authenticate using a client ID:
+
+    ```json
+    {
+        "credential": "managedidentity",
+        "clientId":  "<clientId>"
+    }
+    ```
+
+1. Authenticate using a resource ID:
+
+    ```json
+    {
+        "credential": "managedidentity",
+        "resourceId":  "<managedIdentityResourceId>"
+    }
+    ```
+
+    The resource ID takes the form `/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ManagedIdentity/userAssignedIdentities/{identityName}`.
+
+#### System-assigned identities
+
+1. Authenticate using a client ID:
+
+    ```json
+    {
+        "credential": "managedidentity",
+        "clientid":  "<clientId>"
+    }
+    ```
+
+### Create a `WorkloadIdentityCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.WorkloadIdentityCredential?displayProperty=fullName>:
+
+```json
+{
+    "credential": "workloadidentity",
+    "tenantId":  "<tenantId>",
+    "clientId":  "<clientId>",
+    "tokenFilePath": "<tokenFilePath>"
+}
+```
+
+### Create a `ClientSecretCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ClientSecretCredential?displayProperty=fullName>:
+
+```json
+{
+    "tenantId":  "<tenantId>",
+    "clientId":  "<clientId>",
+    "tokenFilePath": "<clientSecret>"
+}
+```
+
+### Create a `ClientCertificateCredential` type
+
+Add the following configuration values to your _appsettings.json_ file to create an <xref:Azure.Identity.ClientCertificateCredential?displayProperty=fullName>:
+
+```json
+{
+    "tenantId":  "<tenantId>",
+    "clientId":  "<clientId>",
+    "clientCertificate": "<clientCertificate>"
```

Comment:
Should we also document the optional store name and location - https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/extensions/Microsoft.Extensions.Azure/src/Internal/ClientFactory.cs#L95-L96

Reply:
Yes, we should